src/main/resources/kweb/kweb_bootstrap.js

```diff
@@ -203,6 +203,62 @@ function removeClass(el, className) {
     }
 }
 
+class DiffPatchData {
+    constructor(prefixEndIndex, postfixOffset, diffString) {
+        this.prefixEndIndex = prefixEndIndex;
+        this.postfixOffset = postfixOffset;
+        this.diffString = diffString;
+    }
+}
+
+function get_diff_changes(id) {
+    let e = document.getElementById(id);
+    let newString = e.dataset.value;//reads the new string value from data-attribute data-vale
+    let oldString = e.dataset.previousInput;//reads the oldString from the data-attribute data-previous-input
+
+    //This seems like a logical place to do this, but I'm unsure if it is safe to do so
+    //I think it might prematurely change the value of toBind.value in prelude.kt's setValue function
+    //savePreviousInput(newString, id)//put the newString into the data attribute so it can be used as the oldString the next time this method is run
+
+    if (oldString === undefined) {//the first time this is run previous-input should be undefined so we just return the new string
+        return new DiffPatchData(0, 0, newString);
+    }
+    let oldArray = Array.from(oldString);
+    let newArray = Array.from(newString);
+
+    let commonPrefixEnd = 0;
+
+    let oldStringLastIndex = oldArray.length - 1;
+    let newStringLastIndex = newArray.length - 1;
+
+    let commonPostfixOffset = 0;
+
+    let shorterStringLength = (oldArray.length > newArray.length) ? newArray.length : oldArray.length;
+
+    for (let i = 0; i < shorterStringLength; i++) {
+        if (oldArray[i] == newArray[i]) {
+            commonPrefixEnd = i+1;
+        } else break;
+    }
+    for(let offset = 0; offset < shorterStringLength - commonPrefixEnd; offset++) {
+        if (oldArray[oldStringLastIndex - offset] == newArray[newStringLastIndex - offset]) {
+            commonPostfixOffset = offset;
+        } else break;
+    }
+    if (oldStringLastIndex > newStringLastIndex) {//if oldString is longer we are removing text and we send an empty string as the diff
+        return new DiffPatchData(commonPrefixEnd, commonPostfixOffset, "");
+    } else {
+        return new DiffPatchData(commonPrefixEnd, commonPostfixOffset,
+            newString.substring(commonPrefixEnd, newString.length - commonPostfixOffset));
+    }
+}
+
+
+function savePreviousInput(previousInputString, id) {
```

Comment:
Looks like this is unused now.

Reply:
I put that there for use in the get_diff_changes() function. I commented out the part where I call it though, because I wasn't 100% sure it was safe to do so though. I was worried about prematurely changing the data-attribute that stored the previous input.